src/runtime/server/utils/session.ts

```diff
@@ -37,7 +55,16 @@ export async function getUserSession(event: H3Event) {
 export async function setUserSession(event: H3Event, data: UserSession, config?: Partial<SessionConfig>) {
   const session = await _useSession(event, config)
 
-  await session.update(defu(data, session.data))
+  const sessionStorage = getSessionStorage()
+  if (sessionStorage) {
+    await sessionStorage.setItem(`nuxt-session:${session.id}`, {
+      ...data,
+      lastAccess: Date.now(),
+    })
```

Comment:
We should get the session too here as this is extending the session data normally using `defu`

Reply:
I've now extended the session using defu according to your comment https://github.com/atinux/nuxt-auth-utils/pull/265/commits/f8ce2ffdc7a274ba942de8e685d01d02ca8a01c4